diff --git a/test/test_roscpp/test/src/service_exception.cpp b/test/test_roscpp/test/src/service_exception.cpp
index 4b0a0a93a..5a2a97bf9 100644
--- a/test/test_roscpp/test/src/service_exception.cpp
+++ b/test/test_roscpp/test/src/service_exception.cpp
@@ -43,7 +43,7 @@ public:
 protected:
   std::list<log4cxx::spi::LoggingEventPtr> list;
 };
-typedef log4cxx::helpers::ObjectPtrT<ListAppender> ListAppenderPtr;
+LOG4CXX_PTR_DEF(ListAppender);
 
 static const char EXCEPTION[] = "custom exception message";
 
@@ -64,7 +64,7 @@ TEST(roscpp, ServiceThrowingException)
   ros::ServiceServer service = n.advertiseService(SERVICE, throwingService);
 
   log4cxx::LoggerPtr logger = log4cxx::Logger::getLogger("ros.roscpp");
-  ListAppenderPtr appender = new ListAppender();
+  ListAppenderPtr appender(new ListAppender());
   logger->addAppender(appender);
 
   ros::ServiceClient client = n.serviceClient<std_srvs::Empty>(SERVICE, true);
